x/cronos/keeper/evm_test.go

```diff
@@ -0,0 +1,68 @@
+package keeper_test
+
+import (
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/tharsis/ethermint/crypto/ethsecp256k1"
+)
+
+func (suite *KeeperTestSuite) TestDeployContract() {
+	suite.SetupTest()
+	keeper := suite.app.CronosKeeper
+
+	_, err := keeper.DeployModuleCRC20(suite.ctx, "test")
+	suite.Require().NoError(err)
+}
+
+func (suite *KeeperTestSuite) TestTokenConversion() {
+	suite.SetupTest()
+	keeper := suite.app.CronosKeeper
+
+	// generate test address
+	priv, err := ethsecp256k1.GenerateKey()
+	suite.Require().NoError(err)
+	address := common.BytesToAddress(priv.PubKey().Address().Bytes())
+	cosmosAddress := sdk.AccAddress(address.Bytes())
+
+	denom := "ibc/testdenom"
+	amount := big.NewInt(100)
+	coins := sdk.NewCoins(sdk.NewCoin(denom, sdk.NewIntFromBigInt(amount)))
+
+	// mint native tokens
+	err = suite.MintCoins(sdk.AccAddress(address.Bytes()), coins)
+	suite.Require().NoError(err)
```

Comment:
maybe check also user balance

Reply:
It only calls cosmos-sdk's methods, I think we can trust that? 😄